### ISSUE TITLE
Base heat capacity in Li Song is always 2 below the actual capacity of a mech

### ISSUE THREAD

**Issue by NavidA1** [lisongmechlab/lsml#767]
It seems that the base heat capacity that is reported in Li Song is always 2 below what it should be.

Base heat cap of a mech is always 30. The first 10 double heatsinks each add 2.0 capacity on top (2.25 for singles) and after ten, double heatsinks add 0.5 each (0.85 for singles)

Test case:
1- Take a Jenner-IIC-2 (comes with 5% heat quirk) with no skills
2- equip an XL100, 6 extra heatsinks for a total of 10, and two ERPPCs

In MWO: 54% heatspike which is (2 * 14.5 * 0.95) / 51

In Li Song: (2 * 14.5 * 0.95) / 48 = 57%

**Comment by LiSongMWO** [comment]
For my personal notes:

Example heatsink entry in XML (changed since I wrote the original code):

```
  <Module id="3001" name="DoubleHeatSink_MkI" CType="CHeatSinkStats" faction="InnerSphere">
    <ModuleStats slots="3" tons="1" health="7.5" />
    <HeatSinkStats cooling="0.22" engineCooling="0.22" heatbase="-0.5" engineHeatbase="-2.0" />
  </Module>
```

Change [src/main/java/org/lisoft/lsml/model/item/HeatSink.java](https://github.com/lisongmechlab/lsml/blob/23792ad6a451a4eed0a396d97dd416e723e18868/src/main/java/org/lisoft/lsml/model/item/HeatSink.java) to accept the new `engineHeatbase` attribute and use that in [HeatCapacity.java](https://github.com/lisongmechlab/lsml/blob/23792ad6a451a4eed0a396d97dd416e723e18868/src/main/java/org/lisoft/lsml/model/metrics/HeatCapacity.java#L49) instead of the hard coded 20 capacity from enigen heat sinks which was the style at the time. The method also has to be updated to account for cases where engine heat sink are less than 10.

TODO:
1)  ~replicate experiment to see that moving no longer reduces heat base from  #764.~
2) ~devise experiment to see if map heat still affects heat base or if it's only sinking capability.~ It doesn't
3) ~find out of heat capacity quirks and skills still only apply to the base heat, or if it applies to the total heat cap.~ Applies to base + 10 mandatory heatsinks.
4) ~update unit tests~

**Comment by LiSongMWO** [comment]
@NavidA1 can you please verify that `develop` does what you expect it to? I believe it should based on my testing but please confirm.

Git clone, make sure `java` is on your `PATH` and run `gradlew.bat run`  from `cmd.exe` or PowerShell if you prefer.

**Comment by NavidA1** [comment]
I have java jdk installed, yet I'm getting compile errors.

**Comment by LiSongMWO** [comment]
Without knowing the errors or specifics of your set up, I'm going to hazard a guess: You need to use JDK8 from Oracle (not OpenJDK) because of how they changed the packaging of JavaFX between Java 8 and Java 9. 

In Java 8, JavaFX was a part of the Java Runtime and any application could use it without additional dependencies (as LSML does). But in Java 9 this bundling was removed and JavaFX became a [separate project, OpenJFX](https://openjfx.io/), but Java 9 also introduced modular Java which forces you to use the new Jigsaw module system. I've tried and failed on two occasions to upgrade LSML past Java 9 but both times I've ran into a wall where XStream (the XML parsing library that LSML's entire game data and config/save file parsing is based on) is incompatible with Java 9 because of the new security model introduced with Jigsaw. I cannot use OpenJFX because they require Java 11 or later as well.

So basically, because LSML uses Java 8 features and cannot upgrade past Java 9 until the entire game-data parsing code is re-written, and the unbundling of JavaFX is Java 9, and minimum Java version of OpenJFX, you have to use JDK8 for LSML.

Sorry for the confusion, I know this is annoying, trust me, I'm as upset as you are about it.

**Comment by NavidA1** [comment]
Sorry for the late reply.

I can confirm that the develop branch has the correct heat capacity values now. And can be used as a reference for https://github.com/lisongmechlab/lsml/issues/764

**Comment by LiSongMWO** [comment]
Thanks. I'll consider this closed then.
